### PR TITLE
add new components

### PR DESCRIPTION
**Pull request by krunal200**

### CODE REVIEW

File: src/utility.js [hunk 1]
```diff
@@ -13,4 +13,43 @@ function isNumber(n) {
 
 function getConfigValue(config, key, defaultValue) {
 	return config.hasOwnProperty(key) ? config[key] : defaultValue;
+}
+
+function getDomain (url) {
+	var regex = /^(?:(?:https?:)?\/\/)?(?:www\.)?(.*?)\//;
```
Comment (mannanali413) [inline src/utility.js:19]: can we use `window.location.hostname` here?
https://stackoverflow.com/questions/6725890/location-host-vs-location-hostname-and-cross-browser-compatibility

File: src/utility.js [hunk 2]
```diff
@@ -13,4 +13,43 @@ function isNumber(n) {
...
+			try {
+				return getCreativeDuration(ad) + aggregatedTime;
+			} catch (e) {
+				return aggregatedTime;
+			}
+	    }, 0)
+}
+
+function getCreativeDuration (ad) {
+	return ad.creatives[0].duration;
```
Comment (mannanali413) [inline src/utility.js:54]: Can it ever happen that ad.creatives[0] is `null` or `undefined`?

File: src/skin.js [hunk 3]
```diff
@@ -50,35 +53,39 @@ var onPlayerEvent = function(event, eventData) {
...
+            this._controls.progressBar.setProgress(eventData.duration, eventData.position);
+            this._controls.videoStartCountdown.updateTime(eventData.position);
             if(!this._isPlayingVPAID) {
                 this._controls.skipCountdown.update(eventData.position);
             }
             break;
         case OO.Pulse.AdPlayer.Events.LINEAR_AD_PAUSED:
             if(!this._isPlayingVPAID) {            
-                this.setControls([ 'muteButton', 'adCounter', 'progressBar', 'playButton' ]);
+                this.setControls([ 'muteButton', 'adCounter', 'progressBar', 'playButton', 'hoverOverlay', 'videoStartCountdown', 'clickThroughLink', 'controlContainer', 'fullScreenButton' ]);
```
Comment (mannanali413) [inline src/skin.js]: Maybe we can have something like this . 
```
var commonControls = ['muteButton', 'adCounter', 'progressBar', 'videoStartCountdown', 'fullScreenButton', 'controlContainer', 'clickThroughLink']
```
then in the place of setting controls we do
```
this.setControls(commonControls.concat(['hoverOverlay']))
```

File: src/utility.js (hunk 1 above)
Comment (krunal200) [reply src/utility.js:19]: we need this utility to get the hostname from click-through URL and it's not same as user's current domain.

File: src/utility.js (hunk 2 above)
Comment (krunal200) [reply src/utility.js:54]: we always have creative, if the ad is getting played.

File: src/skin.js (hunk 3 above)
Comment (krunal200) [reply src/skin.js]: we don't have common controls across all events.

**Comment by mannanali413** [comment]
cool! let's merge this and publish it

**Pull request merged**